### DISCUSSION TITLE
Cloud Firestore: Does this plugin use firebase offline-persistence by default?

### DISCUSSION THREAD

**Discussion by folsze** [capawesome-team/capacitor-firebase#744] [Q&A]
Afaik firebase has the option to enable offline-persistence which caches requests (& queries afaik) to collections on the client.

Does this plugin make use of this by default automatically? I did not find any options to setup offline persistence but I do want to use it.

I saw it here, only the clearPersistence method:

https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/firestore#clearpersistence

So I assume this does support it. If it does I will consider in the future replacing my whole firebase access with this plugin instead of angular/fire. That's why I am asking before going in the direction of that change.

**Comment by robingenz** [comment]
Yes, this plugin supports offline persistence. From the [Firebase documentation](https://firebase.google.com/docs/firestore/manage-data/enable-offline):

> For Android and Apple platforms, offline persistence is enabled by default. For the web, offline persistence is disabled by default. To enable persistence, call the `enablePersistence` method.

For the web, there is no such `enablePersistence` method yet. Feel free to create a feature request for it.
